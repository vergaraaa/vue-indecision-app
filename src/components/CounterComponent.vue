<template>
  <section>
    <h3>Counter: {{ counter }}</h3>
    <h3>Square: {{ squareCounter }}</h3>

    <div>
      <button class="btn" @click="counter--">-1</button>
      <button class="btn" @click="counter++">+1</button>
    </div>
  </section>
</template>

<script setup lang="ts">
import { useCounter } from '@/composables/useCounter';

const props = defineProps<Props>();

interface Props {
  value: number;
}

const { counter, squareCounter } = useCounter(13);
</script>

<style scoped>
.btn {
  @apply p-5 bg-blue-500 rounded mr-2 hover:bg-blue-700;
}
</style>
