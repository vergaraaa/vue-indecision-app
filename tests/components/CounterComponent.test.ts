import { mount } from '@vue/test-utils';

import CounterComponent from '@/components/CounterComponent.vue';
import { describe, test, expect, it } from 'vitest';

describe('<CounterComponent />', () => {
  test('should match snapshot', () => {
    const wrapper = mount(CounterComponent, {
      props: {
        value: 13,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(CounterComponent, {
      props: {
        value: value,
      },
    });

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );
  });

  test('increment counter when +1 button is clicked', async () => {
    const value = 5;
    const wrapper = mount(CounterComponent, {
      props: {
        value: value,
      },
    });

    const [, btnIncrement] = wrapper.findAll('button');
    await btnIncrement.trigger('click');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value + 1}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${(value + 1) * (value + 1)}`,
    );
  });

  test('decrements counter when -1 button is clicked twice', async () => {
    const value = 5;
    const wrapper = mount(CounterComponent, {
      props: {
        value: value,
      },
    });

    const btnDecrement = wrapper.find('button');
    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value - 2}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${(value - 2) * (value - 2)}`,
    );
  });
});
