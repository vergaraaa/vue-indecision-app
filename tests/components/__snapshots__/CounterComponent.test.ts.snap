// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CounterComponent /> > should match snapshot 1`] = `
"<section data-v-ef93c3cf="">
  <h3 data-v-ef93c3cf="">Counter: 13</h3>
  <h3 data-v-ef93c3cf="" data-testid="square-label">Square: 169</h3>
  <div data-v-ef93c3cf=""><button data-v-ef93c3cf="" class="btn">-1</button><button data-v-ef93c3cf="" class="btn">+1</button></div>
</section>"
`;
